src/hotspot/share/runtime/arguments.cpp

```diff
@@ -2593,9 +2593,20 @@ jint Arguments::parse_each_vm_init_arg(const JavaVMInitArgs* args, JVMFlagOrigin
       } else if (strcmp(tail, ":disable") == 0) {
         LogConfiguration::disable_logging();
         ret = true;
-      } else if (strcmp(tail, ":async") == 0) {
-        LogConfiguration::set_async_mode(true);
+      } else if (strncmp(tail, ":async", strlen(":async")) == 0) {
         ret = true;
+        const char* async_tail = tail + strlen(":async");
+         if (*async_tail == '\0') {
+          // Default is to drop.
+          LogConfiguration::set_async_mode(LogConfiguration::AsyncMode::Drop);
+        } else if (strcmp(async_tail, ":stall") == 0) {
+          LogConfiguration::set_async_mode(LogConfiguration::AsyncMode::Stall);
+        } else if (strcmp(async_tail, ":drop") == 0) {
+          LogConfiguration::set_async_mode(LogConfiguration::AsyncMode::Drop);
+        } else {
+          // User provided unknown async option
+          ret = false;
+        }
```

Comment:
I wonder if parsing the tail should be done by a `LogConfiguration` method instead of exposing the details to the argument parsing code directly?